specs/vep-2448-vdk-run-logs-simplified-and-readable/README.md

```diff
@@ -315,58 +315,106 @@ Explain how does the system handle API violations.
-      running user workloads?
-    * Diagnostics: What are the useful log messages and their required logging
-      levels that could help debug the issue?
-    * Testing: Are there any tests for failure mode? If not, describe why._
-### Operability
-  * What are the SLIs (Service Level Indicators) an operator can use to determine the health of the system.
-  * What are the expected SLOs (Service Level Objectives).
-### Test Plan
-  * Unit tests are expected. But are end to end test necessary. Do we need to extend vdk-heartbeat ?
-  * Are there changes in CICD necessary
-### Dependencies
-  * On what services the feature depends on ? Are there new (external) dependencies added?
-### Security and Permissions
-  How is access control handled?
-  * Is encryption in transport supported and how is it implemented?
-  * What data is sensitive within these components? How is this data secured?
-      * In-transit?
-      * At rest?
-      * Is it logged?
-  * What secrets are needed by the components? How are these secrets secured and attained?
--->
 
+### Log Structure
+
+Users should be able to override the default logging format structure so that
+metadata fields they don't care about are hidden. This should happen through
+standard vdk configuration, e.g. config.ini or env variables. Users should be
+able to pass something like
+
+```
+[vdk]
+log_config=[timestamp, vdk_tag, line_number]
+```
+
+in config.ini. Operators should also be able to pass the same kind of config
+when deploying control service.
+
+https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/helm_charts/pipelines-control-service/values.yaml#L324
+
+The user configuration should take precedence over the global one.
+
+User should be able to do log.info("xxx") inside a data job and vdk
+automatically add structure around it.
+
+Plugins should be able to declare and inject new metadata fields, e.g. DAGs
+should be able to add a dag_name field or a current_job field
+
+The above requirements are more than likely achievable using structlog.
+
+https://www.structlog.org/en/stable/
+
+### Log Less
+
+We should make sure that logs throughout vdk have the correct log level. Any
+logs that don't should be moved to the appropriate level, e.g. we seem to have
+lots of info logs that should be on the debug level.
+
+Users have the ability to to change the log level for specific modules in their
+data job with the `log_level_module` option. We should add documentation around
+this.
+
+### Clean Error Handling
+
+Error messages should clearly state the problem without placeholder and repeated
+logs text so that users can directly understand what went wrong. Users should be
+able to see the original exception when it's passed up the call stack and is in
+the user code so they can handle it.
+
+VDK developers should be discouraged to use generic error messages like "An
+error occurred". This will give more meaningful feedback to users.
```

Comment:
Another idea potenitally. 

For some common errors we can consider creating an error dictionary/repository that stores all the known errors, their descriptions, and potential solutions.

Reply:
Added to https://github.com/vmware/versatile-data-kit/issues/2571